pkg/parser/parser.go

```diff
@@ -109,5 +120,35 @@ func (p *Parser) ParseFromFile(path string) (*Result, error) {
 		return nil, err
 	}
 
+	result, err := p.Parse(string(config))
+	if err != nil || len(result.Errors) != 0 {
+		return result, err
+	}
+
+	// Get the contents of files that might influence the parser results
+	//
+	// For example, when using Dockerfile as CI environment feature[1], the unique hash of the container
+	// image is calculated from the file specified in the "dockerfile" field.
+	//
+	// [1]: https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment
+	filesContents := make(map[string]string)
+	for _, task := range result.Tasks {
+		inst, err := instance.NewFromProto(task.Instance, []*api.Command{})
+		if err != nil {
+			continue
+		}
+		prebuilt, ok := inst.(*instance.PrebuiltInstance)
+		if !ok {
+			continue
+		}
+		contents, err := ioutil.ReadFile(filepath.Join(filepath.Dir(path), prebuilt.Dockerfile))
+		if err != nil {
+			return nil, fmt.Errorf("%w: %v", ErrFilesContents, err)
+		}
+		filesContents[prebuilt.Dockerfile] = string(contents)
+	}
+
+	// Parse again with the file contents supplied
+	p.FilesContents = filesContents
 	return p.Parse(string(config))
```

Comment:
You can skip it `filesContents` is empty and return previously parsed tasks.

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/33/commits/8f96b9ed4f1592eb7677b6f5a4c6e8e186258e20.